app/src/main/java/org/ole/planet/myplanet/ui/courses/TakeCourseFragment.kt

```diff
@@ -77,12 +77,7 @@ class TakeCourseFragment : Fragment(), ViewPager.OnPageChangeListener, View.OnCl
 
         currentStep = getCourseProgress()
 
-        position = when {
-            position > 0 -> position - 1
-            currentStep > 0 -> currentStep - 1
-            else -> 0
-        }
-
+        position = if (currentStep > 0) currentStep - 1 else 0
```

Comment:
this change seems identical to the code removed just in a more simplified logic, does not really improve functionality or fix the issue as shown in the video attached unless I am missing something
[finish button.webm](https://github.com/user-attachments/assets/4f62bc72-a489-4225-9592-fb3aecbcbe14)

Reply:
Hi @Okuro3499 , when we navigate to suveys or take tests, and return back to the TakeCourseFragment, the position was getting updated again. Therefore, this logic prevents the position for getting updated if its already greater than 0 and sync up with currentStep.